zipkin-storage/influxdb/src/main/java/zipkin2/storage/influxdb/InfluxDBSpanConsumer.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2015-2017 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.storage.influxdb;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.influxdb.InfluxDBException;
+import org.influxdb.dto.Point;
+import zipkin2.Annotation;
+import zipkin2.Call;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+import zipkin2.storage.SpanConsumer;
+import org.influxdb.dto.BatchPoints;
+
+final class InfluxDBSpanConsumer implements SpanConsumer {
+  final InfluxDBStorage storage;
+
+  InfluxDBSpanConsumer(InfluxDBStorage influxDB) {
+    this.storage = influxDB;
+    try {
+      storage.get().createDatabase(storage.database());
+    } catch(InfluxDBException e){}
+  }
+
+  @Override
+  public Call<Void> accept(List<Span> spans) {
+    if (spans.isEmpty()) return Call.create(null);
+
+    BatchPoints batch = BatchPoints
+      .database(storage.database())
+      .retentionPolicy(storage.retentionPolicy())
+      .build();
+    for (Span span : spans) {
+      Point point = Point
+        .measurement(storage.measurement())
+        .tag("trace_id", span.traceId())
+        .tag("id", span.id())
+        .tag("parent_id", span.parentId() == null ? span.id() : span.parentId())
+        .tag("name", span.name())
+        .tag("service_name", serviceName(span))
+        .addField("duration_ns", span.duration() * 1000)
+        .time(span.timestamp(), TimeUnit.MICROSECONDS)
+        .build();
+      batch.point(point);
+
+      for (Annotation anno : span.annotations()) {
+        Point annoPoint = Point
+          .measurement(storage.measurement())
+          .tag("trace_id", span.traceId())
+          .tag("id", span.id())
+          .tag("parent_id", span.parentId() == null ? span.id() : span.parentId())
+          .tag("name", span.name())
+          .tag("service_name", serviceName(span))
+          .tag("endpoint_host", host(span))
+          .tag("annotation", anno.value())
+          .addField("duration_ns", span.duration() * 1000)
+          .time(span.timestamp(), TimeUnit.MICROSECONDS)
+          .build();
+        batch.point(annoPoint);
+      }
+
+      if (span.tags() != null) {
+        for (Map.Entry<String, String> tag : span.tags().entrySet()) {
+          Point taggedPoint = Point
+            .measurement(storage.measurement())
+            .tag("trace_id", span.traceId())
```

Comment:
Just a heads up:

> "Tags containing highly variable information like UUIDs, hashes, and random strings will lead to a large number of series in the database, known colloquially as high series cardinality. High series cardinality is a primary driver of high memory usage for many database workloads."

https://docs.influxdata.com/influxdb/v1.3/concepts/schema_and_data_layout/#don-t-have-too-many-series

Reply:
Hello @fmachado thanks for your feedback. Cardinality is something that we are aware of. With the new TSI we will handle a high cardinality and `trace_id` needs to be indexed because it is used as lookup value during a lot of operation. Tracing from a storage point of view is a complicated and big uses cases :+1: